cumulus/parachains/common/src/xcm_config.rs

```diff
@@ -78,3 +78,60 @@ impl<Location: Get<MultiLocation>> ContainsPair<MultiAsset, MultiLocation>
 		matches!(asset.id, Concrete(ref id) if id == origin && origin == &Location::get())
 	}
 }
+
+/// Accepts an asset if it is a native asset from the system (Relay Chain or system parachain).
+pub struct ConcreteNativeAssetFromSystem;
+impl ContainsPair<MultiAsset, MultiLocation> for ConcreteNativeAssetFromSystem {
+	fn contains(asset: &MultiAsset, origin: &MultiLocation) -> bool {
+		log::trace!(target: "xcm::contains", "ConcreteNativeAssetFromSystem asset: {:?}, origin: {:?}", asset, origin);
+		let parent = MultiLocation::parent();
+		let is_system = match origin {
+			// The Relay Chain
+			MultiLocation { parents: 1, interior: Here } => true,
+			// System parachain
+			// TODO: reuse SystemParachains matcher when https://github.com/paritytech/polkadot-sdk/pull/1234 is merged
+			MultiLocation { parents: 1, interior: X1(Parachain(id)) } => *id < 2000,
+			// Others
+			_ => false,
+		};
+		matches!(asset.id, Concrete(id) if id == parent && is_system)
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use super::{
+		ConcreteNativeAssetFromSystem, ContainsPair, GeneralIndex, Here, MultiAsset, MultiLocation,
+		PalletInstance, Parachain, Parent,
+	};
+
+	#[test]
+	fn native_asset_from_sibling_system_para_works() {
+		let expected_asset: MultiAsset = (Parent, 1000000).into();
+		let expected_origin: MultiLocation = (Parent, Parachain(1999)).into();
+
+		assert!(ConcreteNativeAssetFromSystem::contains(&expected_asset, &expected_origin));
+	}
+
+	#[test]
+	fn native_asset_from_sibling_system_para_fails_for_wrong_asset() {
+		let unexpected_assets: Vec<MultiAsset> = vec![
+			(Here, 1000000).into(),
+			((PalletInstance(50), GeneralIndex(1)), 1000000).into(),
+			((Parent, Parachain(1000), PalletInstance(50), GeneralIndex(1)), 1000000).into(),
+		];
+		let expected_origin: MultiLocation = (Parent, Parachain(1000)).into();
+
+		unexpected_assets.iter().for_each(|asset| {
+			assert!(!ConcreteNativeAssetFromSystem::contains(asset, &expected_origin));
+		});
+	}
+
+	#[test]
+	fn native_asset_from_sibling_system_para_fails_for_wrong_origin() {
```

Comment:
I woudl maybe join `native_asset_from_sibling_system_para_works` and `native_asset_from_sibling_system_para_fails_for_wrong_origin` in one test with test data and expected test result.
We can cover more cases with just one test this way, something like this:

```
let test_data = vec![
			(0, true),
			(1, true),
			(1999, true),
			(2000, false),
			(2001, false),
		];

for (para_id, expected_result) in test_data {
		let origin: MultiLocation = (Parent, Parachain(1999)).into();
        assert_eq!(expected_result, ConcreteNativeAssetFromSystem::contains(&expected_asset, &origin));
}

```